### PR TITLE
make justification-period configurable

### PR DESCRIPTION
**Pull request by jasl**
- add `justification_period` to `sc_service::Configuration`
- add `justification_period` to `sc_cli::Configuration` and set default to `512` which is the previous hard-coded value so it should not break or change any behavior
- add `justification-period` to CLI arg

### CODE REVIEW

File: client/cli/src/commands/run_cmd.rs [hunk 1]
```diff
@@ -50,6 +50,11 @@ pub struct RunCmd {
 	#[structopt(long)]
 	pub no_grandpa: bool,
 
+	/// Set the the maximum RPC payload size for both requests and responses (both http and ws), in
```
Comment (xlc) [inline client/cli/src/commands/run_cmd.rs:53]: comment needs updaing

Comment (jasl) [reply client/cli/src/commands/run_cmd.rs]: Oh, my bad, thank you!

**Review by bkchr — CHANGES_REQUESTED** [review]
If you need this for your chain, I propose to make this configurable in your chain. Not really required to update Substrate for this.

**Comment by jasl** [comment]
> If you need this for your chain, I propose to make this configurable in your chain. Not really required to update Substrate for this.

We're parachain, for this part we rely on Polkadot, and Polkadot depends on `sc_cli`, it's difficult to change this in our side.

Besides, there is [a TODO in Polkadot to made it configurable](https://github.com/paritytech/polkadot/blob/master/node/service/src/lib.rs#L869) this PR is partially work

**Comment by h4x3rotab** [comment]
We need our full nodes to save block justification for the relay chain at every block. The full node is introduced by cumulus, which relies on the original polkadot-service. So we wanted to change Substrate first and add a PR on Polkadot side.

**Comment by andresilva** [comment]
This change makes me a bit suspicious about how you're using GRANDPA. Just to make sure we're on the same page, setting this to 1 does not guarantee that every block will have a justification. Although that will be the case for 99% of the time, GRANDPA can still skip blocks while finalizing.

**Comment by h4x3rotab** [comment]
@andresilva Yes, you are right about the justification generation. We believe our use case is valid. Let me add some background here.

On our network we have some off-chain components called `pRuntime`. It's similar to the Substrate OCW. It read all the inputs from the blockchain, and execute some additional logic based on the on-chain storage and events. However, we cannot use OCW directly, because we have to put `pRuntime` inside a special hardware secure enclaves to make sure it's trustworthy and privacy-preserving. ​So we got the architecture:


```text
Substrate --> relayer --> [pRuntime]
```

In this design, from `pRuntime`'s perspective, any data from the outside must be validated. Event the admin of the computer cannot be trusted. So most importantly, it must validate everything from the blockchain before using it.

In our case, `pRuntime` receives batches of blocks while it's running, and it checks the GRANDPA block justification of the last block at every batch, to make sure it's on a finalized chain. In theory we only need the latest block justification in the initial block syncing. However when it's already synced, we need to check the justification of every new block to follow the blockchain.

We don't need every block to have block justifications. However, since `pRuntime` is a realtime service, and we absolute don't want it to wait for the next justified block for a long time (like 512 blocks or so), when a justification is missing.

Therefore when we are running a standalone blockchain, we just set it to 1, and it works well. But in the context of parachain, we need to relay chain client to produce as many block justification as possible, to minimize the delay of block syncing process.

**Comment by h4x3rotab** [comment]
Oh, totally missed `grandpa_proveFinality`. Checking it out now.

**Comment by jasl** [comment]
a dumb question: 
I just try `grandpa_proveFinality`, it always raise error like `-32602: Invalid params: invalid type: string "0x0000000000000000000000000000000000000000000000000000000000000001", expected u32.
it's weird `Hash` should be `H256` not `u32`

I test this RPC on Kusama

**Pull request closed without merging**